### PR TITLE
v2.0.0

### PR DESCRIPTION
**Pull request by jklenzing**
Summary
 - New Features
   - `pysatData` directory created in user's home directory if no directory specified
   - Added preliminary support for `xarray` to the `instrument` object
   - Support for `today`, `tomorrow`, and `yesterday` as datetime objects
   - Added `model_utils`, featuring preliminary support for data-model comparison
   - Added support for 1d median in seasonal averages
   - Added routine to convert from kp to Ap
   - Added `pyglow` integration support for python 3.x
   - Improves input handling for datetime parameters that are more precise than just year, month, and day, where appropriate
   - Added merging routines to allow combination of measured and forecasted Kp and F10.7 indexes into a single instrument object
 - Code restructure
   - Moved instrument templates and methods to subdirectories
   - Moved utils into multiple subdirectories to aid with organization
 - Instrument Updates
   - NASA CDAWeb download now uses https protocol rather than FTP
   - `_instrument.py` supports xarray
   - Support for listing files from remote server
   - COSMIC RO data unified into single instrument object
   - Added support for DEMETER IAP
   - Added support for DMSP IVM Level 1 data
   - Added routines to update DMSP ephemeris and drifts
   - Added warnings to instruments without download support
   - Added support for ICON FUV and MIGHTI
   - Added support for JRO ISR
   - Added support for F10.7 and more Kp forecast products
   - Added instrument templates for Madrigal, CDAWeb, and netcdf_pandas
   - Added support for TIMED SABER
   - Added support for UCAR TIEGCM
   - Switched download methods for CDAWeb and COSMIC data to use `requests`
   - Removed support for SuperDARN and SuperMAG downloads while server changes are sorted out
 - Updates to travis configuration
   - Tests run for python 2.7 and 3.7
   - Added display port to test plots
 - Updates to community docs
   - Added Issue templates
   - Added Pull Request Template
   - Added note for PR to be made to develop, not master
 - Style updates throughout
   - Consistent documentation for docstrings and instruments
   - Cleaned up commented code lines
   - PEP8 scrub
 - Documentation
   - Added FAQ section
   - Added "powered by pysat" logo
   - Updated supported instruments
 - Unit Test Updates
   - Dropped instrument templates from coverage
   - Added multiple output options for `pysat_testing` object to aid with constellation tests. Removed old constellation test objects.
   - Added test data for space weather indices to speed up testing
   - Cyclic data for test instruments now generated from single test method
   - test objects for xarray added
   - Added test for parsed delimited files
   - Removed ftp downloads from travis tests, still will run locally
 - Bug fixes
   - `pandas.ix` notation replaced with `pandas.loc` and `pandas.iloc` throughout
   - Fixed a bug that forced user into interactive mode in `ssnl.plot`
   - Bug fixes and cleanup in demo codes
   - Fix for orbit iteration when data is not loaded
   - Fixed a bug in python 3.7 caused by change in behaviour of StopIteration (#207)
   - Update to use of `len` on xarray to handle new behaviour (#130)
   - Updated import of reload statements now that python 3.3 has reached end of life
   - Updated deprecated behaviour of `get_duplicates`, `.apply`, and `.to_csv` when using pandas
   - Fixed bug in assigning units to metadata (#162)
   - Fixed timing bug introduced by reading only the first date/data pair from each line in the 45-day file data blocks

### CODE REVIEW

File: pysat/_instrument.py [hunk 1]
```diff
@@ -118,44 +122,44 @@ class Instrument(object):
...
         ivm.download(start,stop)
         ivm.load(2009,1)
         print(ivm['ionVelmeridional'])
-        
+
         # Ionosphere profiles from GPS occultation
-        cosmic = pysat.Instrument('cosmic2013', 
-                                    'gps', 
-                                    'ionprf', 
+        cosmic = pysat.Instrument('cosmic2013',
```
Comment (jklenzing) [inline pysat/_instrument.py]: Not consistent with current use of cosmic data

File: pysat/instruments/superdarn_grdex.py [hunk 2]
```diff
@@ -43,11 +44,103 @@
...
+    # How to acknowledge use of SuperDARN Data - 2017
+    print('Authors should acknowledge the use of SuperDARN data. ',
+          'SuperDARN is a collection of radars funded by national scientific ',
+          'funding agencies of Australia, Canada, China, France, Italy, ',
+          'Japan, Norway, South Africa, United Kingdom and the United States ',
+          'of America.')
+    return
+
+
+def list_remote_files(tag, sat_id, data_path=None, format_str=None):
```
Comment (jklenzing) [inline pysat/instruments/superdarn_grdex.py]: Why is this still here?

File: pysat/instruments/supermag_magnetometer.py [hunk 3]
```diff
@@ -17,54 +17,167 @@
...
+    # reset the list_remote_files routine to include the data path
+    # now conveniently included with instrument object
+    self._list_remote_rtn = \
+        functools.partial(list_remote_files,
+                          data_path=self.files.data_path,
+                          format_str=self.files.file_format)
+    return
+
+
+def list_remote_files(tag='', sat_id=None, data_path=None, format_str=None,
```
Comment (jklenzing) [inline pysat/instruments/supermag_magnetometer.py]: Likewise this?  Do these routines even work now that download is not an option?

File: pysat/instruments/superdarn_grdex.py (hunk 2 above)
Comment (jklenzing) [reply pysat/instruments/superdarn_grdex.py]: OK, routine does not access remote data.  Leaving as is.

File: pysat/instruments/supermag_magnetometer.py (hunk 3 above)
Comment (jklenzing) [reply pysat/instruments/supermag_magnetometer.py]: same as above

File: pysat/_instrument.py (hunk 1 above)
Comment (jklenzing) [reply pysat/_instrument.py]: fixed in #250.

**Comment by jklenzing** [comment]
The summary above is also the changelog text added in #250.

**Comment by jklenzing** [comment]
> Ran across a few things that probably should be fixed before the merge. Thoughts?

Never mind.

File: pysat/instruments/dmsp_ivm.py [hunk 4]
```diff
@@ -23,172 +23,170 @@
...
 platform = 'dmsp'
 name = 'ivm'
-tags = {'utd':'UTDallas DMSP data processing'}
-sat_ids = {'f11':['utd'], 'f12':['utd'], 'f13':['utd'], 'f14':['utd'], 'f15':['utd']}
-test_dates = {'f11':{'utd':pysat.datetime(1998,1,2)},
-              'f12':{'utd':pysat.datetime(1998,1,2)},
-              'f13':{'utd':pysat.datetime(1998,1,2)},
-              'f14':{'utd':pysat.datetime(1998,1,2)},
-              'f15':{'utd':pysat.datetime(2017,12,30)}}
+tags = {'utd': 'UTDallas DMSP data processing', '': 'Level 1 data processing'}
```
Comment (jklenzing) [inline pysat/instruments/dmsp_ivm.py]: Is the new data Level 1 or 2?

Comment (rstoneback) [reply pysat/instruments/dmsp_ivm.py]: level 2, see pull #250

**Comment by jklenzing** [comment]
OK, everything is merged in here.  Let me know if you catch anything else.

**Pull request merged**